solo-chains/runtime/starlight/src/lib.rs

```diff
@@ -1521,6 +1525,57 @@ impl pallet_registrar::RegistrarHooks for StarlightRegistrarHooks {
     }
 }
 
+pub struct BabeSlotBeacon;
+
+impl BlockNumberProvider for BabeSlotBeacon {
+    type BlockNumber = u32;
+
+    fn current_block_number() -> Self::BlockNumber {
+        // TODO: nimbus_primitives::SlotBeacon requires u32, but this is a u64 in pallet_babe, and
+        // also it gets converted to u64 in pallet_author_noting, so let's do something to remove
+        // this intermediate u32 conversion, such as using a different trait
+        u64::from(pallet_babe::CurrentSlot::<Runtime>::get()) as u32
+    }
+}
+
+impl pallet_author_noting::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type ContainerChains = ContainerRegistrar;
+    type SlotBeacon = BabeSlotBeacon;
+    type ContainerChainAuthor = TanssiCollatorAssignment;
+    // We benchmark each hook individually, so for runtime-benchmarks this should be empty
+    #[cfg(feature = "runtime-benchmarks")]
+    type AuthorNotingHook = ();
+    #[cfg(not(feature = "runtime-benchmarks"))]
+    type AuthorNotingHook = ();
+    // TODO: uncomment when pallets exist
+    //type AuthorNotingHook = (InflationRewards, ServicesPayment);
+    type RelayOrPara = pallet_author_noting::RelayMode;
+    type WeightInfo = pallet_author_noting::weights::SubstrateWeight<Runtime>;
+}
+
+pub struct NoRelayChainStateProvider;
```

Comment:
We should move this to primitives.

Also there is still a public getter, I would assume if they remove the getter, they would make the storage public.

But in Addition, you can rely on this trait https://github.com/paritytech/polkadot-sdk/blob/02e50adf7ba6cc65a9ef5c332b3e2974c8d23f48/substrate/frame/system/src/lib.rs#L2187C17-L2187C36

Reply:
This was needed for `type RelayChainStateProvider = ...`, but since I moved that to be inside of `ParaMode`, it is no longer needed here, I can just remove it.